x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/nlp/tokenizers/DebertaV2Tokenizer.java

```diff
@@ -0,0 +1,306 @@
+            }
+            return sortedVocab;
+        }
+
+        public DebertaV2Tokenizer.Builder setNeverSplit(Set<String> neverSplit) {
+            this.neverSplit = neverSplit;
+            return this;
+        }
+
+        public DebertaV2Tokenizer.Builder setMaxSequenceLength(int maxSequenceLength) {
+            this.maxSequenceLength = maxSequenceLength;
+            return this;
+        }
+
+        /**
+         * Include CLS and SEP tokens
+         * @param withSpecialTokens if true include CLS and SEP tokens
+         * @return this
+         */
+        public DebertaV2Tokenizer.Builder setWithSpecialTokens(boolean withSpecialTokens) {
+            this.withSpecialTokens = withSpecialTokens;
+            return this;
+        }
+
+        public DebertaV2Tokenizer build() throws IOException {
+            if (neverSplit == null) {
+                neverSplit = Collections.emptySet();
+            }
+
+            return new DebertaV2Tokenizer(originalVocab, vocab, scores, withSpecialTokens, maxSequenceLength, neverSplit);
+        }
+    }
+
+    @Override
+    public InnerTokenization innerTokenize(String seq) {
+        List<Integer> tokenPositionMap = new ArrayList<>();
+        try (TokenStream ts = debertaAnalyzer.tokenStream("input", seq)) {
+            ts.reset();
+            PositionIncrementAttribute tokenPos = ts.addAttribute(PositionIncrementAttribute.class);
+            int currPos = -1; // the PositionIncrement starts at one, so this aligns the first token at position 0
+            while (ts.incrementToken()) {
+                currPos += tokenPos.getPositionIncrement();
+                tokenPositionMap.add(currPos);
+            }
+        } catch (IOException ex) {
+            throw new UncheckedIOException(ex);
+        }
+        return new InnerTokenization(new ArrayList<>(debertaAnalyzer.getTokens()), tokenPositionMap);
+    }
+
+    @Override
+    public void close() {
+        this.debertaAnalyzer.close();
+    }
+
+    static class DebertaAnalyzer extends Analyzer {
+        private final List<String> vocabulary;
+        private final List<String> neverSplit;
+        private final double[] scores;
+        private UnigramTokenizer innerTokenizer;
+        private final String unknownToken;
+        private final PrecompiledCharMapNormalizer.Config normalizer;
+
+        DebertaAnalyzer(List<String> vocabulary, List<Double> scores, List<String> neverSplit, String unknownToken) throws IOException {
+            this.vocabulary = vocabulary;
+            this.neverSplit = neverSplit;
+            this.unknownToken = unknownToken;
+            this.scores = new double[scores.size()];
+            int i = 0;
+            for (Double s : scores) {
+                this.scores[i++] = s;
+            }
+            normalizer = PrecompiledCharMapNormalizer.fromBase64EncodedResource(
+                "/org/elasticsearch/xpack/ml/inference.nlp.tokenizers/spm_precompiled_normalizer.txt"
```

Comment:
This file is the sentence piece model for the XLM Roberta model added in #94089. Doesn't DeBERTa use a different model. If so we either have to include it as a resource or add the ability to install it

Reply:
I'll need to verify this in the end to end testing. I had tried to extract the normalizer from the spm file provided in huggingface but I wasn't able to do it. I'm also not sure that the normalizer is different.

Reply:
I haven't been able to complete the end-to-end testing yet because the cpp snapshot isn't available yet. I created a notebook to test it though.

Reply:
I just went through the steps of extracting the precompiled charsmap from the deberta SPM file and it is identical to this older one